### PR TITLE
Fix time display #911

### PR DESCRIPTION
**Pull request by eshifri**
This PR fixies incorrect current time display.
Issue #911

### CODE REVIEW

**Comment by eshifri** [comment]
Turned out Statistics was also affected.

**Comment by pfeerick** [comment]
Yeah, I think that was intentional (part of https://github.com/EdgeTX/edgetx/pull/896), since it clearly denotes whether a measure of hours or minutes, etc, but I'm not fussed either way how it looks since IMO if hours is a possibility for statistics, it should have three lots of digits.

**Comment by pfeerick** [comment]
No it's not used as a boolean. `getTimerString()` is an overloaded function:

https://github.com/EdgeTX/edgetx/blob/69a465f0e267edc6ebe1f830ec31010f6acfe244/radio/src/strhelpers.h#L49-L51

https://github.com/EdgeTX/edgetx/blob/69a465f0e267edc6ebe1f830ec31010f6acfe244/radio/src/strhelpers.h#L68-L69

And this is having basically no effect atm, since you haven't considered what is actually going on here yet ;)

https://github.com/EdgeTX/edgetx/blob/69a465f0e267edc6ebe1f830ec31010f6acfe244/radio/src/strhelpers.cpp#L745-L758

In other words, unless another function is added, or perhaps the function changed so it does accept more than just boolean as a selector of the separator character, it isn't (and doesn't) look any different yet. ;)

**Comment by pfeerick** [comment]
> `char *getTimerStringFormat(int32_t tme, uint8_t uFormat ); ` uFormat is a enum
> 
> for example : {UNIT_UPPERCASE, UNIT_LOWERCASE, UNIT_COMMA} and the number of digit group

That was pretty much exactly what I had in mind. So now the question is how to go about this... ie. keeping building on this PR, cherrypick into a new PR, or start from main again? I'm hinting here I would *like* to come back in a day or so and find something that I can poke holes in and test... however it happens ;)

**Comment by eshifri** [comment]
BTW, I think it works as you want it to
screen-2021-10-17-101330
screen-2021-10-17-101336
:

**Comment by eshifri** [comment]
#896 provides automatic timer data format and this PR provides distinguishing between TIMER and TIME.

**Comment by Runsys16** [comment]
Hi,

I found 10 files where getTimerString__() functions are used.

```
./radio/src/strhelpers.cpp	   171	char *getTimerStringCase(char *dest, int32_t tme, uint8_t hours,
./radio/src/lua/api_model.cpp	   231	@function model.getTimer(timer)
./radio/src/lua/api_colorlcd.cpp	   370	  getTimerString(s, tme, (flags & TIMEHOUR) != 0);
./radio/src/strhelpers.h	    49	char *getTimerString(char *dest, int32_t tme, uint8_t hours = 0u);
./radio/src/gui/colorlcd/view_statistics.cpp	    97	      [] { return getTimerString(sessionTimer, TIMEHOUR); },
./radio/src/gui/colorlcd/draw_functions.cpp	   427	  getTimerString(str, tme, (flags & TIMEHOUR) != 0);
./radio/src/gui/colorlcd/layouts/topbar_impl.cpp	    95	  getTimerString(str, getValue(MIXSRC_TX_TIME));
./radio/src/gui/colorlcd/timeedit.cpp	    45	               getTimerString(getValue(), (textFlags & TIMEHOUR) != 0),
./radio/src/gui/colorlcd/themes/480_default.cpp	   388	      getTimerString(str, getValue(MIXSRC_TX_TIME));
./radio/src/gui/colorlcd/themes/flysky.cpp	   406	      getTimerString(str, getValue(MIXSRC_TX_TIME));
```

I propose to rewrite the code for having only one function char *getTimerStringFormat(int32_t tme, uint8_t uFormat )"
I can do it , I'm free for today. (If you want)
But I create a new PR for creating a new git branch.

**Comment by eshifri** [comment]
> So why is the time still formatted as 10h13m with this PR? 😆 Or have I missed something entirely (brain freeze day)...

Since the current time is (was) 10h13m (10:13). Without this PR it shows 10m13s.

**Comment by pfeerick** [comment]
> Since the current time is (was) 10h13m (10:13). Without this PR it shows 10m13s.

 Oops... so it does! Looks I should be wearing my glasses... 😮😆 I'll just stand over here in the corner for a while then! 😁

**Comment by eshifri** [comment]
Please don't! Yourtime is too important to spend it in the corner! :-)

**Comment by pfeerick** [comment]
I only glanced at the commit you did yesterday, and that was already starting to look good, so I'll have a look properly tonight at the latest changes. I'd suggest getTimerStringCase can get renamed to something more descriptive since it is more than just upper/lowercase now... perhaps something getFormattedTimerString or getTimerStringFormatted or something else... whatever fits in with the naming conversion in use...

**Comment by eshifri** [comment]
I'm not sure about switching between "h" and ":"; not that I'm against it - just have doubts.
Another tricky part is how to decide the number of digital groups to show: two ot three.
#896  provided automatic solution, bu it does not take into account the size (of the widget). And in general  it is nice solution, since it always work and does not require additional thinking when using it. :-)

**Comment by pfeerick** [comment]
What I was suggesting would still be fully automatic.... Just that it would only resort to trickery like `MM:SS/HHhMM` if a 'compact-form' flag was passed, otherwise it would stick with `HH:MM:SS` until > 24 hours, then move to `1d HH:MM`.

**Comment by eshifri** [comment]
And "compact-form" is a caller decision? 
That will work.
